pkg/dashboards/dashboards.go

```diff
@@ -226,12 +226,25 @@ func dashboardNameExists(dname string, orgid uint64) bool {
 	return exists
 }
 
+var defaultDashboardNames = map[string]bool{
+	"Siglens Query DB":     true,
+	"Siglens Ingestion DB": true,
+	"Siglens Data DB":      true,
+	"Sample Dashboard":     true,
+}
```

Comment:
It would be good to remove this duplication. As I'm sure you know, these names are defined in allids.json:
```
{
    "10329b95-47a8-48df-8b1d-0a0a01ec6c42": "Siglens Ingestion DB",
    "a28f485c-4747-4024-bb6b-d230f101f852": "Siglens Query DB",
    "bd74f11e-26c8-4827-bf65-c0b464e1f2a4": "Siglens Data DB",
    "53cb3dde-fd78-4253-808c-18e4077ef0f1": "Sample Dashboard"
}
```
With the changes you have, it's fairly likely that later we would add/rename/delete a default dashboard and just update allids.json but forget to update this map you've defined, and then we'll get basically the bug you're fixing now.

I think a better approach would be that on startup (so we only have to do it once), you parse allids.json to get the names of the default dashboards and store those. That way if we add/rename/delete a default dashboard in a later release, we'll only have to update one thing instead of two things.

Side note: it looks like you're using this `map` as a `set`. Golang doesn't have a `set`, but you can use `map[string]struct{}` like a set. This has the benefit of being a little more clear for your use case—it also uses slightly less memory, but that doesn't matter for this particular case.

Reply:
So I should do the parsing in the startup.go file and not on dashboard.go?

Reply:
No, the function to setup this map/set should be in dashboards.go. To get it to run on startup, you could either:
- Make it a public function, and call it inside startup.go
- Call it from `func init() { ... }` in dashboards.go. Golang automatically runs all `init()` functions on startup, as long as the `main` package includes the specified package (in this case `dashboards`). So as long as `startup.go` includes `dashboards` either directly or indirectly, the `init()` inside `dashboards` will get run.

Reply:
I have been testing my changes and when I test it manually, and by this I mean opening localhost and trying to create a dashboard with a default name by clicking in the New Dashboard Button, it seems to be working as intended as it does brings the error message and does not create the dashboard. But whenever i run make test,  the Test_Dashboard_storage_methods fails due to the InitDashboards function returning (level=error msg="InitDashboard: Failed to read default dashboards file, err=open defaultDBs/allids.json: no such file or directory") and i cant seem to find a fix for it, i dont think its because of the path since both dashboards.go and dashboards_test.go are in the same folder
![image_2024-07-22_191426292](https://github.com/user-attachments/assets/db0e7ef9-5efa-44d5-861a-e647f778b25c)

Reply:
I think the tests get run from within the package. You can test this with something like:
```
pwd := os.Getenv("PWD")
log.Errorf("(testing) InitDashboard: PWD=%v", pwd)
```
One option is that for the unit test you could write the file (in the directory the tests are run in) before calling `InitDashboards()`, and then remove the file at the end of the test.

Alternatively, maybe you can find a good way to read from the real `defaultDBs/allids.json` file